backend/server.py

```
@@ -1,50 +1,98 @@
+        USE_CELERY_INFERENCE = False
+
+# Load models directly if not using Celery inference (backward compatibility)
+if not USE_CELERY_INFERENCE:
+    logger.info("⚠️  Running in LEGACY mode - loading models directly in Flask process")
+    logger.info("   This uses more memory. Consider setting USE_CELERY_INFERENCE=true")
+    MCQGen = main.MCQGenerator()
+    answer = main.AnswerPredictor()
+    BoolQGen = main.BoolQGenerator()
+    ShortQGen = main.ShortQGenerator()
+    qg = main.QuestionGenerator()
+    qa_model = pipeline("question-answering")
```

Ruff (0.15.7)

[error] Undefined name `pipeline`

(F821)